From 54f7d6829d853c0e7af221c9d1c62527320f95c0 Mon Sep 17 00:00:00 2001
From: Oleksandr Shamray <oleksandrs@nvidia.com>
Date: Tue, 15 Jun 2021 16:47:15 +0300
Subject: [PATCH] Fix cpld*_pn reading

---
 drivers/platform/mellanox/mlxreg-io.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/platform/mellanox/mlxreg-io.c b/drivers/platform/mellanox/mlxreg-io.c
index e3c0d4f..c199d6d 100644
--- a/drivers/platform/mellanox/mlxreg-io.c
+++ b/drivers/platform/mellanox/mlxreg-io.c
@@ -101,10 +101,8 @@ mlxreg_io_get_reg(void *regmap, struct mlxreg_core_data *data, u32 in_val,
 			ret = regmap_read(regmap, data->reg + i, &val);
 			if (ret)
 				goto access_error;
-
-			*regval |= rol32(val, regsize * i);
+			*regval |= rol32(val, regsize * i * 8);
 		}
-		*regval = le32_to_cpu(*regval & regmax);
 	}
 
 access_error:
-- 
2.8.4

