--- a/drivers/pci/pci.c	2025-10-22 15:29:42.937891072 +0300
+++ b/drivers/pci/pci.c	2025-10-22 15:32:41.866608014 +0300
@@ -4860,7 +4860,7 @@
 static bool pcie_wait_for_link_delay(struct pci_dev *pdev, bool active,
 				     int delay)
 {
-	int timeout = 1000;
+	int timeout = 20000;	/* Debug change to 20sec. */
 	bool ret;
 	u16 lnk_status;
 
@@ -5026,7 +5026,7 @@
 			delay);
 		if (!pcie_wait_for_link_delay(dev, true, delay)) {
 			/* Did not train, no need to wait any further */
-			pci_info(dev, "Data Link Layer Link Active not set in 1000 msec\n");
+			pci_info(dev, "Data Link Layer Link Active not set in 20000 msec\n");
 			return -ENOTTY;
 		}
 	}
